go/vt/key/key.go

```diff
@@ -45,6 +48,12 @@ func (i Uint64Key) Bytes() []byte {
 	return buf
 }
 
+type DestinationTarget struct {
```

Comment:
This name is confusing. How about `QualifiedDestination`?

Reply:
I'm actually wondering if we even need this type. Could we get away with always using the relevant individual fields, without the structure? One reason I'm saying that is for instance the TabletType is not useful in delete.go below, as a delete would always be on the master.

The parse method can look like:
parse(toParse string, defaultType tabletType) (keyspace string, tabletType, Destination, error)

Reply:
@alainjobart  - I originally went down this approach but thought it was too verbose when we actually need the all the values: 

```
keyspace, tabletType, dest,  err := key.ParseDestination(targetString, defaultTabletType)
```

I leaned towards the struct because it was more concise.

Reply:
Another reason for this is we pass keysapce/tablettype + single Destination, or keyspace/tablettype + []Destination in a few methods. Like in srvtoporesolver.go:

func (r *Resolver) ResolveDestinations(ctx context.Context, keyspace string, tabletType topodatapb.TabletType, ids []*querypb.Value, destinations []key.Destination) ([]*ResolvedShard, [][]*querypb.Value, error) {

I agree it's more concise with the structure, but then we'd need another one with []Destination too, and it's starting to bloat the code...

(note I'm not opposed to thi that much, it just feels overkill)

Reply:
I think it's better to not have the type (for now). If the combination of vars gains widespread usage, we should look at making it a core type used everywhere.

Not having the type also avoids the argument about what to name it :).

Reply:
These are good points! 

I updated to not have this data structure.